pages/api/auth/[...nextauth].js

```diff
@@ -0,0 +1,108 @@
+import NextAuth from 'next-auth'
+import Providers from 'next-auth/providers'
+import ldap from 'ldapjs'
+
+const AD = require('activedirectory2').promiseWrapper;
+
+
+const url = process.env.LDAP_URL
+if (!url) throw new Error('LDAP URL missing. Add it via env variables')
+const baseDn = process.env.BASE_DN
+if (!baseDn) throw new Error('Base DN missing.')
+const ldapIdField = process.env.LDAP_MAPPING_UID || 'uid'
+const ldapNameField = process.env.LDAP_MAPPING_NAME || 'displayName'
+const ldapEmailField = process.env.LDAP_MAPPING_MAIL || 'mail'
+const ldapAdminUsername = process.env.LDAP_ADMIN_USERNAME || ''
+const ldapAdminPassword = process.env.LDAP_ADMIN_PASSWORD || ''
+
+const site = process.env.SITE || 'http://localhost:3000'
+
+const client = ldap.createClient({
+  url
+})
+
+const config = { url: url,
+               baseDN: baseDn,
+               username: ldapAdminUsername,
+               password: ldapAdminPassword }
+const ad = new AD(config);
+
+
+const options = {
+    site,
+    secret: process.env.JWT_SECRET,
+    session: {
+      jwt: true,
+      secret: process.env.JWT_SECRET
+    },
+    jwt: {
+    },
+    debug: true,
+    providers: [
+        Providers.Credentials({
+          // The name to display on the sign in form (e.g. 'Sign in with...')
+          name: 'LDAP',
+          // The credentials is used to generate a suitable form on the sign in page.
+          // You can specify whatever fields you are expecting to be submitted.
+          // e.g. domain, username, password, 2FA token, etc.
+          credentials: {
+            username: { label: "Username", type: "text", placeholder: "jsmith" },
+            password: {  label: "Password", type: "password" }
+          },
+          authorize: async (credentials) => {
+        
+            let user = null
+
+          
+            let username = `cn=${credentials.username},${baseDn}`
```

Comment:
@JannikZed Please make the `usernameAttribute` configurable. The default should be `uid` instead of `cn`. See https://www.kimai.org/documentation/ldap.html